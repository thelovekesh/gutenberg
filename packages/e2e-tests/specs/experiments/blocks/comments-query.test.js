/**
 * WordPress dependencies
 */
import {
	insertBlock,
	createNewPost,
	activateTheme,
	getEditedPostContent,
	clickBlockAppender,
} from '@wordpress/e2e-test-utils';
/**
 * WordPress dependencies
 */

// It might make sense to include a similar test in WP core (or move this one over).
// See discussion here: https://github.com/WordPress/gutenberg/pull/32797#issuecomment-864192088.
describe( 'Comment Query Loop', () => {
	beforeAll( async () => {
		await activateTheme( 'emptytheme' );
		// await setOption( 'page_comments', true, 'options-discussion.php' );
		await createNewPost();
	} );
	it( 'Comment Query block insertion', async () => {
		// We won't select the option that we updated and will also remove some
		// _transient options that seem to change at every update.
		await insertBlock( 'Comments Query Loop' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
	it( 'can be created by typing "/comment query loop"', async () => {
		// Create a list with the slash block shortcut.
		await clickBlockAppender();
		await page.keyboard.type( '/comments query loop' );
		await page.waitForXPath(
			`//*[contains(@class, "components-autocomplete__result") and contains(@class, "is-selected") and contains(text(), 'Comments Query Loop')]`
		);
		await page.keyboard.press( 'Enter' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
	afterAll( async () => {
		await activateTheme( 'twentytwentyone' );
	} );
} );
