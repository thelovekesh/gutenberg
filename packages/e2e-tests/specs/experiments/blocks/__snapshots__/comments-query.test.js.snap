// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Query Loop Comment Query block insertion 1`] = `
"<!-- wp:comments-query-loop -->
<div class=\\"wp-block-comments-query-loop\\"><!-- wp:comment-template /-->

<!-- wp:comments-pagination -->
<!-- wp:comments-pagination-previous /-->

<!-- wp:comments-pagination-numbers /-->

<!-- wp:comments-pagination-next /-->
<!-- /wp:comments-pagination --></div>
<!-- /wp:comments-query-loop -->"
`;

exports[`Comment Query Loop can be created by typing "/comment query loop" 1`] = `
"<!-- wp:comments-query-loop -->
<div class=\\"wp-block-comments-query-loop\\"><!-- wp:comment-template /-->

<!-- wp:comments-pagination -->
<!-- wp:comments-pagination-previous /-->

<!-- wp:comments-pagination-numbers /-->

<!-- wp:comments-pagination-next /-->
<!-- /wp:comments-pagination --></div>
<!-- /wp:comments-query-loop -->

<!-- wp:comments-query-loop -->
<div class=\\"wp-block-comments-query-loop\\"><!-- wp:comment-template /-->

<!-- wp:comments-pagination -->
<!-- wp:comments-pagination-previous /-->

<!-- wp:comments-pagination-numbers /-->

<!-- wp:comments-pagination-next /-->
<!-- /wp:comments-pagination --></div>
<!-- /wp:comments-query-loop -->"
`;
